virtio-devices/src/device.rs

```diff
@@ -286,8 +286,17 @@ impl VirtioCommon {
     pub fn wait_for_epoll_threads(&mut self) {
         if let Some(mut threads) = self.epoll_threads.take() {
             for t in threads.drain(..) {
-                if let Err(e) = t.join() {
-                    error!("Error joining thread: {:?}", e);
+                let mut counter = 0;
+                while counter < 5 {
+                    if t.is_finished() {
+                        if let Err(e) = t.join() {
+                            error!("Error joining thread: {:?}", e);
+                        }
+                        break;
+                    } else {
+                        std::thread::sleep(std::time::Duration::from_millis(500));
+                        counter += 1;
+                    }
```

Comment:
I think @likebreath might have changed the design a few times, and at some point the following lines were lost:
```rust
if let Some(kill_evt) = self.kill_evt.take() {
            // Ignore the result because there is nothing we can do about it.
            let _ = kill_evt.write(1);
}
```
Each fuzzer ensures some specific eventfds are triggered, but then they expect the thread to terminate because a kill_evt is triggered.
I'd advise to wait for @likebreath to review this PR as he might give us some better feedback about what's intended.

Reply:
You can't just fire the kill event because then the thread wouldn't get round to actually executing on the queue event.

Reply:
I agree, but at some point I thought @likebreath introduced a design with some sort of steps validation allowing to be sure the queue_evt would always be processed before the kill_evt would.

Reply:
The fuzzer was designed to not use timeout for two reasons: 1) it makes fuzzing slower (as the fuzzer would have to wait for the fixed amount of 'timeout' to finish even the worker thread is finished faster) and hence less efficient ; 2) it masks potential real slow or hanging cases.

Let me take a closer look at hanging on rng and balloon reported from oss-fuzz and see if I can fix them without the need of a fixed timeout.

Reply:
When do you expect the threads to exit? If doesn't happen on receipt of a write to`exit_evt` because it doesn't do that. AFAICT they would only exit if the running of the event loop resulted in an error. If the data injected by the fuzzer is fine then the thread will continue.

Reply:
The worker thread will return by itself when there is no events to process, which is why I needed to provide a custom `EpollHelper::run()` to achieve that behavior: 
https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/virtio-devices/src/epoll_helper.rs#L245-L247.

Reply:
Ah, okay.